bundles/org.openhab.binding.neato/src/main/java/org/openhab/binding/neato/internal/handler/NeatoHandler.java

```diff
@@ -146,7 +145,9 @@ private void publishChannels() {
         updateProperty(Thing.PROPERTY_MODEL_ID, neatoState.getMeta().getModelName());
 
         updateState(CHANNEL_STATE, new StringType(neatoState.getRobotState().name()));
-        updateState(CHANNEL_ERROR, new StringType((String) ObjectUtils.defaultIfNull(neatoState.getError(), "")));
+
+        String error = neatoState.getError() != null ? (String) neatoState.getError() : "";
```

Comment:
The cast to String is apparently useless as `neatoState.getError()` returns a String

Reply:
True, guess i mixed it up with the original line. anyway, fixed.